File: Irrgarten-ruby/Irrgarten/weapon.rb
# frozen_string_literal: true

module Irrgarten
  require_relative 'dice'
  require_relative 'combat_element'

class Weapon < CombatElement

  def initialize(e,u)
    super(e,u)
  end

  def attack
    produce_effect
  end

  def to_string
    "W[#{super.to_string}"
  end
  public_class_method :new
end
end

File: Irrgarten-Java/src/irrgarten/Weapon.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

/**
 *
 * @author dpant
 */
public class Weapon extends CombatElement{
    
    Weapon (float power_, int uses_){ 
       super(power_,uses_);
    }  
    
    public float attack (){
        return produceEffect();
    }
    @Override
    public String toString(){
       return "W"+super.toString();  
    }
  
}
